applications/ShapeOptimizationApplication/tests/opt_process_vertex_morphing_test/run_test.py

```diff
@@ -24,54 +24,10 @@
 import KratosMultiphysics.kratos_utilities as kratos_utilities
 import csv, os
 
-# =======================================================================================================
-# Define external analyzer
-# =======================================================================================================
-
-class CustomAnalyzer(AnalyzerBaseClass):
-
-    # --------------------------------------------------------------------------------------------------
-    def AnalyzeDesignAndReportToCommunicator(self, current_design, optimization_iteration, communicator):
-        if communicator.isRequestingValueOf("targetDeviation"):
-            communicator.reportValue("targetDeviation", self.__ObjectiveFunction(current_design))
-
-        if communicator.isRequestingGradientOf("targetDeviation"):
-            communicator.reportGradient("targetDeviation", self.__ObjectiveGradient(current_design))
-
-    # --------------------------------------------------------------------------------------------------
-    def __ObjectiveFunction(self, current_design):
-        """ Returns the objective function to be minimized """
-        objective = 0.0
-        for node in current_design.Nodes:
-            objective = objective + abs(self.__TentFunction(node.X) - node.Z)
-        return objective
-
-    # --------------------------------------------------------------------------------------------------
-    def __ObjectiveGradient(self, current_design):
-        """ Returns the gradient of the objective function """
-        sensitivity = dict()
-        for node in current_design.Nodes:
-            delta = node.Z - self.__TentFunction(node.X)
-            if abs(delta) == 0.0:
-                sz = 0.0
-            else:
-                sz = delta / abs(delta)
-            sensitivity[node.Id] = [0.0, 0.0, sz]
-        return sensitivity
-
-    # --------------------------------------------------------------------------------------------------
-    @staticmethod
-    def __TentFunction(x):
-        """ Defines the target curve z=__TentFunction(x) """
-        if x <= 15.0:
-            return 0.0
-        elif x<= 20.0:
-            return (x - 15.0) / 5.0
-        elif x <= 25.0:
-            return 1.0 - (x - 20.0) / 5.0
-        else:
-            return 0.0
-
+try:
+    from .tent_analyzer import CustomAnalyzer
+except ImportError:
+    from tent_analyzer import CustomAnalyzer
```

Comment:
why is this necessary?

Reply:
to be able to run the test file directly - and not from the test suite.